src/domain.rs

```diff
@@ -166,7 +166,8 @@ impl Domain<Global> {
 
 /// Generate a [`Domain`] with an entirely unique domain family.
 ///
-/// The generated family implements [`Singleton`], which enables the use of [`AtomicPtr::safe_load`].
+/// The generated family implements [`Singleton`], which enables the use of
+/// [`crate::AtomicPtr::safe_load`].
```

Comment:
I would just add
```rust
#[cfg(doc)]
use crate::AtomicPtr;
```
near the top of the file. That way it will work as-is. It's a trick we can apply elsewhere in this PR too where we currently have to specify `](crate::`.

Reply:
The issue with `[crate::AtomicPtr]` specifically is that the file already imports `std::sync::atomic::AtomicPtr`. It seems a `#[cfg(doc)] ↵ use crate::*;` does not override the existing binding, so we would need separate `#[cfg(docs)]`+`#[cfg(not(docs))]` `use`s to make it work, which sounds like just shifting the same degree of mess elsewhere.

Good idea for the rest though. I did this in [0167d97](https://github.com/jonhoo/haphazard/pull/29/commits/0167d97bea0db24667d0fac3b888d40b8191a386). For the `crate::AtomicPtr` ones that are necessary as explained above, I hid the `crate::` prefix in the link part only, so it doesn't pollute the visible text.